pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -511,96 +521,80 @@ contract StablePhantomPool is StablePool, ProtocolFeeCache {
-        )
-    {
-        uint256 totalTokens = _getTotalTokens();
-
-        // This join neither grants BPT nor takes any tokens from the sender.
-        bptOut = 0;
-        amountsIn = new uint256[](totalTokens);
-
-        // Due protocol fees are all zero except for the BPT amount, which is then zeroed out.
-        dueProtocolFeeAmounts = new uint256[](totalTokens);
-        dueProtocolFeeAmounts[_bptIndex] = _dueProtocolFeeBptAmount;
-        _dueProtocolFeeBptAmount = 0;
-    }
-
-    /**
-     * @dev Revert on all exits.
+     * @dev Revert on all exit kinds (except for the recovery mode exit, which will not call `_onExitPool`)
      */
     function _onExitPool(
         bytes32,
         address,
         address,
-        uint256[] memory balances,
+        uint256[] memory,
         uint256,
         uint256,
         uint256[] memory,
-        bytes memory userData
+        bytes memory
     )
         internal
-        view
+        pure
         override
         returns (
-            uint256 bptAmountIn,
-            uint256[] memory amountsOut,
-            uint256[] memory dueProtocolFeeAmounts
+            uint256,
+            uint256[] memory,
+            uint256[] memory
         )
     {
-        StablePhantomPoolUserData.ExitKindPhantom kind = userData.exitKind();
-
-        // Exits typically revert, except for the proportional exit when the emergency pause mechanism has been
-        // triggered. This allows for a simple and safe way to exit the Pool.
-        if (kind == StablePhantomPoolUserData.ExitKindPhantom.EXACT_BPT_IN_FOR_TOKENS_OUT) {
-            _ensurePaused();
-
-            // Note that this will cause the user's BPT to be burned, which is not something that happens during
-            // regular operation of this Pool, and may lead to accounting errors. Because of this, it is highly
-            // advisable to stop using a Pool after it is paused and the pause window expires.
-
-            (bptAmountIn, amountsOut) = _proportionalExit(balances, userData);
-            // For simplicity, due protocol fees are set to zero.
-            dueProtocolFeeAmounts = new uint256[](_getTotalTokens());
-        } else {
-            _revert(Errors.UNHANDLED_BY_PHANTOM_POOL);
-        }
+        _revert(Errors.UNHANDLED_BY_PHANTOM_POOL);
     }
 
-    function _proportionalExit(uint256[] memory balances, bytes memory userData)
-        private
-        view
-        returns (uint256, uint256[] memory)
-    {
-        // This proportional exit function is only enabled if the contract is paused, to provide users a way to
-        // retrieve their tokens in case of an emergency.
-        //
-        // This particular exit function is the only one available because it is the simplest, and therefore least
-        // likely to be incorrect, or revert and lock funds.
+    // We cannot use the default implementation here, since we need to account for the BPT token
+    function _doRecoveryModeExit(
```

Comment:
I think there's a simpler way to achieve this and rely more on the base impl, highlighting the differences. Something like:

```
function _doRecoveryModeExit(
    uint256[] memory balances,
    uint256,
    bytes memory userData
) internal virtual override returns (uint256, uint256[] memory) {
    // Since this Pool uses preminted BPT, we need to replace the total supply with the virtual total supply, and
    // adjust the balances array by removing BPT from it.
    (uint256 bptAmountIn, uint256[] memory amountsOut) = super._doRecoveryModeExit(
        _dropBptItem(balances), 
        _getVirtualSupply(balances[_bptIndex])
    );

    return (bptAmountIn, _addBptItem(amountsOut, 0));
}
```